lib/AST/TypeSubstitution.cpp

```diff
@@ -445,10 +445,9 @@ Type TypeSubstituter::transformDependentMemberType(DependentMemberType *dependen
   auto conformance = IFS.lookupConformance(origBase, proto, level);
 
   auto result = conformance.getTypeWitness(assocType, IFS.getOptions());
-  if (result->is<ErrorType>()) {
-    auto substBase = origBase.subst(IFS);
-    return DependentMemberType::get(ErrorType::get(substBase), assocType);
```

Comment:
Turns out this is one place where we were constructing an `ErrorType` with a solver-allocated original type. Changed it to just construct a plain ErrorType instead. Any objections to this @slavapestov? As far as I can tell, the original type is mainly used for debugging purposes. It's also used for a bit of diagnostic logic in `getTypeForDisplay`, but that shouldn't apply here as that's looking at an interface type.

Reply:
Oh oops, the printing logic here is needed for completion...

Reply:
Updated to check for `hasTypeVariableOrPlaceholder` instead